### ISSUE TITLE
Licensing fails when local services are unavailable and not retried

### ISSUE THREAD

**Issue by jgruberf5** [F5Networks/f5-declarative-onboarding#160]
### Environment
 * Declarative Onboarding Version: 1.15.0-3
 * BIG-IP Version: 14.1.2.6-0.0.2.0

### Summary
When attempting to register a license, if the replace call to /tm/shared/licensing/registration does not respond or resets the socket (`tryUntil: got error {"code":"ECONNRESET"}`), a valid declaration is rolled-back and delcarative onboarding returns is standard malformed JSON playload (due to error reporting trunction) to requests.

Access to a failed host is available if requested. 

Circumstances which cause this connection reset appear in about 3% of all of our deployed VE (tested over 2500). 

### Steps To Reproduce
1. Submit the following declaration:
```json
{
	"async": true,
	"schemaVersion": "1.0.0",
	"Common": {
		"dnsServers": {
			"nameServers": ["161.26.0.10", "161.26.0.11"],
			"search": ["local"],
			"class": "DNS"
		},
		"ntpServers": {
			"timezone": "UTC",
			"class": "NTP",
			"servers": ["0.pool.ntp.org", "1.pool.ntp.org"]
		},
		"configsync": {
			"configsyncIp": "x.x.x.x",
			"class": "ConfigSync"
		},
		"hostname": "t-a8ea11ea-e53c-4399-8c61-abb10994704e.local",
		"net_1_2": {
			"interfaces": [{
				"tagged": false,
				"name": "1.2"
			}],
			"tag": 4092,
			"class": "VLAN",
			"mtu": 1500
		},
		"selfip_1_2": {
			"trafficGroup": "traffic-group-local-only",
			"vlan": "net_1_2",
			"allowService": "all",
			"class": "SelfIp",
			"address": "x.x.x.x/xx"
		},
		"selfip_1_3": {
			"trafficGroup": "traffic-group-local-only",
			"vlan": "net_1_3",
			"allowService": "all",
			"class": "SelfIp",
			"address": "x.x.x.x/xx"
		},
		"selfip_1_1": {
			"trafficGroup": "traffic-group-local-only",
			"vlan": "net_1_1",
			"allowService": "all",
			"class": "SelfIp",
			"address": "x.x.x.x/xx"
		},
		"utilityLicense": {
			"licensePool": "XXXXXXX",
			"licenseType": "licensePool",
			"bigIqHost": "x.x.x.x",
			"bigIqUsername": "admin",
			"skuKeyword1": "F5-BIG-MSP-BR-3G",
			"unitOfMeasure": "yearly",
			"bigIqPassword": "xxxxxxx",
			"reachable": false,
			"skuKeyword2": null,
			"hypervisor": "kvm",
			"class": "License"
		},
		"net_1_3": {
			"interfaces": [{
				"tagged": false,
				"name": "1.3"
			}],
			"tag": 4091,
			"class": "VLAN",
			"mtu": 1500
		},
		"class": "Tenant",
		"net_1_1": {
			"interfaces": [{
				"tagged": false,
				"name": "1.1"
			}],
			"tag": 4093,
			"class": "VLAN",
			"mtu": 1500
		},
		"failoverAddress": {
			"class": "FailoverUnicast",
			"address": "x.x.x.x"
		}
	},
	"class": "Device",
	"label": "generated from DHCPv4 metadata"
}
```

2. Observe the following error response:
```json
{"code":500,"message":"request failed with null exception","originalRequestBody":"{\"code\":500,\"message\":\"java.net.ProtocolException: status:500, body:{\\\"id\\\":\\\"764fcf0e-db97-4a3a-9421-412755ae83f5\\\",\\\"selfLink\\\":\\\"https://localhost/mgmt/shared/declarative-onboarding/task/764fcf0e-db97-4a3a-9421-412755ae83f5\\\",\\\"result\\\":{\\\"class\\\":\\\"Result\\\",\\\"code\\\":\\\"ECONNRESET\\\",\\\"status\\\":\\\"ERROR\\\",\\\"message\\\":\\\"invalid config - rolled back\\\",\\\"errors\\\":[\\\"tryUntil: max tries reached: socket hang up\\\",\\\"tryUntil: max tries reached: socket hang up\\\"]},\\\"declaration\\\":{\\\"async\\\":true,\\\"schemaVersion\\\":\\\"1.0.0\\\",\\\"Common\\\":{\\\"dnsServers\\\":{\\\"nameServers\\\":[\\\"161.26.0.10\\\",\\\"161.26.0.11\\\"],\\\"search\\\":[\\\"local\\\"],\\\"class\\\":\\\"DNS\\\"},\\\"ntpServers\\\":{\\\"timezone\\\":\\\"UTC\\\",\\\"class\\\":\\\"NTP\\\",\\\"servers\\\":[\\\"0.pool.ntp.org\\\",\\\"1.pool.ntp.org\\\"]},\\\"configsync\\\":{\\\"configsyncIp\\\":\\\"10.242.65.170\\\",\\\"class\\\":\\\"ConfigSync\\\"},\\\"hostname\\\":\\\"t-a8ea11ea-e53c-4399-8c61-abb10994704e.local\\\",\\\"net_1_2\\\":{\\\"interfaces\\\":[{\\\"tagged\\\":false,\\\"name\\\":\\\"1.2\\\"}],\\\"tag\\\":4092,\\\"class\\\":\\\"VLAN\\\",\\\"mtu\\\":1500,\\\"cmpHash\\\":\\\"default\\\",\\\"failsafeEnabled\\\":false,\\\"failsafeAction\\\":\\\"failover-restart-tm\\\",\\\"failsafeTimeout\\\":90},\\\"selfip_1_2\\\":{\\\"trafficGroup\\\":\\\"traffic-group-local-only\\\",\\\"vlan\\\":\\\"net_1_2\\\",\\\"allowService\\\":\\\"all\\\",\\\"class\\\":\\\"SelfIp\\\",\\\"address\\\":\\\"10.242.66.170/24\\\"},\\\"selfip_1_3\\\":{\\\"trafficGroup\\\":\\\"traffic-group-local-only\\\",\\\"vlan\\\":\\\"net_1_3\\\",\\\"allowService\\\":\\\"all\\\",\\\"class\\\":\\\"SelfIp\\\",\\\"address\\\":\\\"10.242.64.170/24\\\"},\\\"selfip_1_1\\\":{\\\"trafficGroup\\\":\\\"traffic-group-local-only\\\",\\\"vlan\\\":\\\"net_1_1\\\",\\\"allowService\\\":\\\"all\\\",\\\"class\\\":\\\"SelfIp\\\",\\\"address\\\":\\\"10.242.65
```

### Expected Behavior
A valid declaration should be provisioned. We need retries when errors are not with the declaration, but on the BIG-IP.

### Actual Behavior
cloud-libs bigIpOnboard.js  installLicense sets maxRetries to 0 and retryInterval to 0 when calling `/tm/shared/licensing/registration`

**Comment by dstokesf5** [comment]
We have AUTOTOOL-1890 in our internal product backlog to investigate the retry behavior of licensing.

**Comment by dstokesf5** [comment]
This was resolved in the 1.16.0 release.
